### ISSUE TITLE
[java] False positive of AccessorMethodGeneration with same method name

### ISSUE THREAD

**Issue by vanniktech** [pmd/pmd#613]
**Rule Set:** `java/design`

**Description:**

The following code below gets flagged:

```java
private void setVisibility(final int visibility) throws Throwable {
  activityTestRule.runOnUiThread(new Runnable() {
    @Override public void run() {
      activityTestRule.getActivity().imageView.setVisibility(visibility);
    }
  });
}
```

with: 

```
<violation beginline="80" endline="80" begincolumn="49" endcolumn="62" rule="AccessorMethodGeneration" ruleset="Design" package="com.vanniktech.espresso.core.utils" class="DrawableMatcherTest" method="run" externalInfoUrl="https://pmd.github.io/pmd-5.8.1/pmd-java/rules/java/design.html#AccessorMethodGeneration" priority="3">
Avoid autogenerated methods to access private fields and methods of inner / outer classes
</violation>
```

The rule assumes that the `setVisibility` method that's called inside the Runnable is the same as the function that the code is in, hence this gets flagged. My guess is that type resolution is missing.

The moment I apply this diff, the rule won't complain any more:

```diff
-private void setVisibility(final int visibility) throws Throwable {
+void setVisibility(final int visibility) throws Throwable {
```

**Running PMD 5.8.1 through:** Gradle

**Comment by jsotuyod** [comment]
@vanniktech thanks for the report. This is a duplicate of #342, I'll close it to keep track there.

**Comment by vanniktech** [comment]
Whops sorry for the noise then.
